ga4gh/datamodel/datasets.py

```diff
@@ -237,63 +252,80 @@ def __init__(
+    should remove this class and replace the testing data structures with
+    an alternative approach.
     """
     variantsDirName = "variants"
     readsDirName = "reads"
     featuresDirName = "sequenceAnnotations"
 
+    @classmethod
+    def fromRow(cls, row, dataRepository):
+        print("When is this called??", row)
+        dataset = cls(row[b'name'], row[b'dataDir'], dataRepository)
+        dataset._description = row[b'description']
+        return dataset
+
     def __init__(self, localId, dataDir, dataRepository):
         super(FileSystemDataset, self).__init__(localId)
         self._dataDir = dataDir
-        self._setMetadata()
 
         # Variants
         variantSetDir = os.path.join(dataDir, self.variantsDirName)
+        # We need a referenceSet instance for variants and features.
+        # To make this work we just pick the first reference set.
+        # This is NOT a good idea!!
+        # None of this code is intended to last long, just until we get
+        # all our test data into the repo format.
+        referenceSet = dataRepository.getReferenceSets()[0]
         for localId in os.listdir(variantSetDir):
             relativePath = os.path.join(variantSetDir, localId)
             if os.path.isdir(relativePath):
-                variantSet = variants.HtslibVariantSet(
-                    self, localId, relativePath, dataRepository)
+                variantSet = variants.HtslibVariantSet(self, localId)
+                variantSet.populateFromDirectory(relativePath)
+                variantSet.setReferenceSet(referenceSet)
                 self.addVariantSet(variantSet)
-            # Variant annotations sets
-                if variantSet.isAnnotated(relativePath):
+                # Variant annotations sets
+                if variantSet.isAnnotated():
+                    vaName = localId + "_va"  # TODO does this make sense?
+                    sequenceOntology = dataRepository.getOntology(
+                        "sequence_ontology")
                     variantAnnotationSet = variants.HtslibVariantAnnotationSet(
-                            self, localId, relativePath, dataRepository,
-                            variantSet)
+                            variantSet, vaName, sequenceOntology)
                     self.addVariantAnnotationSet(variantAnnotationSet)
-
         # Reads
         readGroupSetDir = os.path.join(dataDir, self.readsDirName)
         for filename in os.listdir(readGroupSetDir):
             if fnmatch.fnmatch(filename, '*.bam'):
                 localId, _ = os.path.splitext(filename)
                 bamPath = os.path.join(readGroupSetDir, filename)
-                readGroupSet = reads.HtslibReadGroupSet(
-                    self, localId, bamPath, dataRepository)
+                readGroupSet = reads.HtslibReadGroupSet(self, localId)
+                readGroupSet.populateFromFile(bamPath, bamPath + ".bai")
+                referenceSet = dataRepository.getReferenceSetByName(
```

Comment:
This is temporary code just to keep our test cases going for now, and is basically just a hack to keep things working. This will all be deleted once we have a SQL repo describing the test data.